rfcs/0001-shabka-usage-design.md

```diff
@@ -0,0 +1,186 @@
+---
+feature: shabka-usage-design
+start-date: 2019-10-14
+author: Marc 'risson' Schmitt
+co-authors:
+related-issues: https://github.com/kalbasit/shabka/projects/10 https://github.com/kalbasit/shabka/issues/263 https://github.com/kalbasit/shabka/issues/246 https://github.com/kalbasit/shabka/issues/55 https://github.com/kalbasit/shabka/issues/281
+---
+
+# Summary
+[summary]: #summary
+
+The goal is to design a system where `shabka` is just a tool like
+`home-manager`, and relies on configuration files provided by the user.
+The tools and ways to configure at the disposition of the user would be those:
+
+* the command `shabka`, installed like `home-manager` which basically does the
+  stuff the user wants to do on a regular basis that we currently have in
+  `./scripts` ;
+* a `dotshabka` like directory for the user's personal preferences settings and
+  default values for `shabka` options, or overriding stuff we define in
+  `shabka`. We will refer to this as `dotshabka-user` ;
+* a `dotshabka` like directory for the user's hosts. We will refer to this as
+  `.shabka`.
+
+# Motivation
+[motivation]: #motivation
+
+This is important in the process of making `shabka` usable for everyone. A
+simple CLI, plus good documentation on how to create configuration files, will
+allow the user to quickly get a hang on the tool.
+
+The planned use cases are very simple for now, and can be described as this:
+
+* Usage on a NixOS system, with root access.
+* Management of several hosts' configuration
+* Management of several users per-host.
+* Management of several users' configuration per-host.
```

Comment:
Do you mind including `* Usage on a Darwin system, with root access.`. The entire paragraph re-written below for clarity.

```
* Usage on a NixOS system, with root access.
** Management of several hosts' configuration
** Management of several users per-host.
** Management of several users' configuration per-host.
* Usage on a Darwin system, with root access.
** Management of several hosts' configuration
** Management of several users per-host.
** Management of several users' configuration per-host.
```